macaroon_service_test.go

```diff
@@ -94,6 +96,122 @@ func TestMacaroonServiceMigration(t *testing.T) {
 	require.NoError(t, ms.Stop())
 }
 
+// TestMacaroonGeneration tests that the macaroon service generates macaroons
+// properly, and doesn't regenerate the macaroon if the required permissions
+// haven't changed, but that it also does regenerate the macaroon when required
+// as the required permissions change.
+func TestMacaroonGeneration(t *testing.T) {
+	// Create a temporary directory where we can store the macaroon db and
+	// the macaroon we are about to create.
+	tempDirPath, err := ioutil.TempDir("", ".testMacaroons")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDirPath)
+
```

Comment:
can replace the whole thing with:

`tempDirPath := t.TempDir()`

Reply:
Ah neat! Thanks for that :)!